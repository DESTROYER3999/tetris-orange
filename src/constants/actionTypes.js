export const JOIN_ROOM = "JOIN_ROOM";
export const UPDATE_ROOM = "UPDATE_ROOM";
export const LEAVE_ROOM = "LEAVE_ROOM";
export const LOBBY_ROOM = "LOBBY_ROOM";

export const NEW_PLAYER = "NEW_PLAYER";
export const DISCONNECT_PLAYER = "DISCONNECT_PLAYER";
export const UPDATE_PLAYER = "UPDATE_PLAYER";
export const UPDATE_PLAYERS_LIST = "UPDATE_PLAYERS_LIST";
export const DELETE_PLAYER = "DELETE_PLAYER";

export const HANDLE_HASH = "HANDLE_HASH";

export const UPDATE_BOARD = "UPDATE_BOARD";
export const UPDATE_GAME = "UPDATE_GAME";
export const UPDATE_GAME_MODE = "UPDATE_GAME_MODE";
export const UPDATE_SPECTRES = "UPDATE_SPECTRES";

export const DISPLAY_TOAST = "DISPLAY_TOAST";

export const CONNECTION = "connection";
export const DISCONNECT = "disconnect";

export const ADD_CHAT_MESSAGE = "ADD_CHAT_MESSAGE";
export const NEW_CHAT_MESSAGE = "NEW_CHAT_MESSAGE";

export const KEY_PRESSED = "KEY_PRESSED";
