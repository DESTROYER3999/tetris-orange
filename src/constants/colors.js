export const MALUS = "#3d3d3d";

export const VIVID_BLUE = "#143cdc";

export const CYAN_BLUE = "#14d4dc";

export const CRIMSON = "#dc143c";

export const PURE_ORANGE = "#ff8b00";

export const LIME_GREEN = "#3cdc14";

export const PURE_VIOLET = "#b600ff";

export const PURE_YELLOW = "#fff400";

export const DARK_CYAN = "#375f64";

export const GREEN_CHAT = "#73F054";

export const RED_CHAT = "#ff3366";
