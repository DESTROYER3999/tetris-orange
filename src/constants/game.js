export const SOLO = "Solo";
export const BATTLEROYAL = "Battleroyal";
export const MAX_PLAYER_SOLO = 1;
export const MAX_PLAYER_BATTLEROYAL = 5;
export const BOARD_HEIGHT = 20;
export const BOARD_WIDTH = 10;
