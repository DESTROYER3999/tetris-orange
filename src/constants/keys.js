export const enumKeys = {
  ARROW_LEFT: "ArrowLeft",
  ARROW_RIGHT: "ArrowRight",
  ARROW_DOWN: "ArrowDown",
  ARROW_UP: "ArrowUp",
  SPACE: "Space",
  ENTER: "Enter"
};
