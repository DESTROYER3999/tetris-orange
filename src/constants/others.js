export const LOG_LINE =
  "\n_________________________________________________________________________________________________________\n\n";
