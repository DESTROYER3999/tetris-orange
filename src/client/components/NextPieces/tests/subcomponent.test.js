import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import NextPiecesSub from "../subcomponent";
import socket from "../../../services/socket-api";
import { VIVID_BLUE } from "../../../../constants/colors";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

const nextPiece = {
  shape: [[0, 0, 0, 0], [1, 1, 1, 1], [0, 0, 0, 0], [0, 0, 0, 0]],
  pos: { x: 3, y: -3 },
  color: VIVID_BLUE
};

describe("<LoginSub />", () => {
  test("NextPiecesSub rendering with a valid nextPiece", () => {
    const wrapper = render(<NextPiecesSub nextPiece={nextPiece} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("NextPiecesSub rendering without nextPiece", () => {
    const wrapper = render(<NextPiecesSub nextPiece={{}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
