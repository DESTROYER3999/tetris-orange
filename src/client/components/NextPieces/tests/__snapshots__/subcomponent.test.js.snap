// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> NextPiecesSub rendering with a valid nextPiece 1`] = `
<div
  class="sc-bdVaJa bKDtjc"
  id="next-pieces-container"
>
  <p>
    NEXT:
  </p>
  <div
    id="next-piece"
  >
    <div
      class="sc-htpNat bmqPRd"
      id="row"
    >
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
    </div>
    <div
      class="sc-htpNat bmqPRd"
      id="row"
    >
      <div
        class="sc-bwzfXH eePTJi"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH eePTJi"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH eePTJi"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH eePTJi"
        color="#143cdc"
      />
    </div>
    <div
      class="sc-htpNat bmqPRd"
      id="row"
    >
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
    </div>
    <div
      class="sc-htpNat bmqPRd"
      id="row"
    >
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
      <div
        class="sc-bwzfXH jIYkEg"
        color="#143cdc"
      />
    </div>
  </div>
</div>
`;

exports[`<LoginSub /> NextPiecesSub rendering without nextPiece 1`] = `
<div
  class="sc-bdVaJa bKDtjc"
  id="next-pieces-container"
>
  <p>
    NEXT:
  </p>
  <div
    id="next-piece"
  />
</div>
`;
